tests/lib/rules/exports-style.js

```diff
@@ -146,14 +156,154 @@ new RuleTester().run("exports-style", rule, {
+            globals: { module: false, exports: true },
+            errors: [
+                "Unexpected access to 'module.exports'. Use 'exports' instead.",
+            ],
+        },
+        {
+            code: "module.exports = { ['a' + 'b']: 1 }",
+            output: "exports['a' + 'b'] = 1;",
+            options: ["exports"],
+            parserOptions: { ecmaVersion: 6 },
+            globals: { module: false, exports: true },
+            errors: [
+                "Unexpected access to 'module.exports'. Use 'exports' instead.",
+            ],
+        },
+        {
+            code: "module.exports = { 'foo': 1 }",
+            output: "exports['foo'] = 1;",
+            options: ["exports"],
+            parserOptions: { ecmaVersion: 6 },
+            globals: { module: false, exports: true },
+            errors: [
+                "Unexpected access to 'module.exports'. Use 'exports' instead.",
+            ],
+        },
+        {
+            code: "module.exports = { foo(a) {} }",
+            output: "exports.foo = function (a) {};",
+            options: ["exports"],
+            parserOptions: { ecmaVersion: 8 },
+            globals: { module: false, exports: true },
+            errors: [
+                "Unexpected access to 'module.exports'. Use 'exports' instead.",
+            ],
+        },
+        {
+            code: "module.exports = { *foo(a) {} }",
+            output: "exports.foo = function* (a) {};",
+            options: ["exports"],
+            parserOptions: { ecmaVersion: 6 },
+            globals: { module: false, exports: true },
+            errors: [
+                "Unexpected access to 'module.exports'. Use 'exports' instead.",
+            ],
+        },
+        {
+            code: "module.exports = { async foo(a) {} }",
+            output: "exports.foo = async function (a) {};",
+            options: ["exports"],
+            parserOptions: { ecmaVersion: 8 },
+            globals: { module: false, exports: true },
+            errors: [
+                "Unexpected access to 'module.exports'. Use 'exports' instead.",
+            ],
+        },
+        {
+            code: "module.exports.foo()",
+            output: "exports.foo()",
+            options: ["exports"],
+            parserOptions: { ecmaVersion: 8 },
+            globals: { module: false, exports: true },
+            errors: [
+                "Unexpected access to 'module.exports'. Use 'exports' instead.",
+            ],
+        },
+        {
+            code: "a = module.exports.foo + module.exports['bar']",
+            output: "a = exports.foo + exports['bar']",
+            options: ["exports"],
+            parserOptions: { ecmaVersion: 8 },
+            globals: { module: false, exports: true },
+            errors: [
+                "Unexpected access to 'module.exports'. Use 'exports' instead.",
+                "Unexpected access to 'module.exports'. Use 'exports' instead.",
+            ],
+        },
```

Comment:
can you add a few more cases for non-top-level exports, e.g.

```js
if(foo){ module.exports = { foo: 1};}
else { module.exports = {foo: 2};}
```

Reply:
Good shout, 8425c8063e3310b19c5c7eeb20f44c6b185ee7cd